File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens) {
        stack<int> bruh;

        for(string c : tokens){
            if(c == "+"){
                int b = bruh.top();
                bruh.pop();
                int a = bruh.top();
                bruh.pop();
                bruh.push(a+b);
            }
            else if(c == "-"){
                int b = bruh.top();
                bruh.pop();
                int a = bruh.top();
                bruh.pop();
                bruh.push(a-b);
            }
            else if(c == "*"){
                int b = bruh.top();
                bruh.pop();
                int a = bruh.top();
                bruh.pop();
                bruh.push(a*b);
            }
            else if(c == "/"){
                int b = bruh.top();
                bruh.pop();
                int a = bruh.top();
                bruh.pop();
                bruh.push(a/b);
            }
            else{
                int num = stoi(c);
                bruh.push(num);
            }
        }
        return bruh.top();
        
    }
};

File: 0150-evaluate-reverse-polish-notation/0150-evaluate-reverse-polish-notation.java
import java.util.Stack;
class Solution {
    public int evalRPN(String[] tokens) 
    {
        Stack <Integer> Stack = new Stack<>();
        for(String token : tokens ){
            if (token.equals("+")||token.equals("*")||token.equals("-")||token.equals("/")){
                int b = Stack.pop();
                int a = Stack.pop();
                if(token.equals("+")){
                    Stack.push(a+b);
                }
                else if (token.equals("-")){
                    Stack.push(a-b);
                }
                  else if (token.equals("*"))
                     {
                       Stack.push(a*b);
                      }
                 else if (token.equals("/")){
                    Stack.push(a/b);
                }
            }
            else{
                Stack.push(Integer.parseInt(token));
             }
        
        }
        return Stack.pop();
    }
    
    
}